scipy/stats/_continuous_distns.py

```diff
@@ -2162,7 +2162,13 @@ def _pdf(self, x, c):
         return np.exp(self._logpdf(x, c))
 
     def _logpdf(self, x, c):
-        return np.log(c) - x - (c+1.0)*sc.log1p(np.exp(-x))
+        # Two mathematically equivalent expressions for log(pdf(x, c)):
+        #     log(pdf(x, c)) = log(c) - x - (c + 1)*log(1 + exp(-x))
+        #                    = log(c) + c*x - (c + 1)*log(1 + exp(x))
+        t = _lazywhere(x > 0, (x, c),
+                       f=lambda x, c: -x - (c + 1)*sc.log1p(np.exp(-x)),
+                       f2=lambda x, c: c*x - (c + 1)*sc.log1p(np.exp(x)))
```

Comment:
Free free to ignore:

I've always assumed `_lazywhere` is fairly slow. which seems to be generally true
```
In [18]: %timeit _lazywhere((t > 0), (t,), lambda v: sc.log1p(v), -np.inf)                                                            
46.3 µs ± 507 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [19]: %timeit sc.log1p(t)                                                                                                          
1.74 µs ± 35.1 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

At the cost of some ugliness it seems we can avoid it here
```
import numpy as np
import scipy.special as sc
from scipy._lib._util import _lazywhere

In [4]: c = 5 
In [5]: x = np.arange(-10, 10, 0.78) 
In [6]: def method1(x, c): 
   ...:    return _lazywhere(x > 0, (x, c), 
   ...:        f=lambda x, c: -x - (c + 1)*sc.log1p(np.exp(-x)), 
   ...:        f2=lambda x, c: c*x - (c + 1)*sc.log1p(np.exp(x))) 
   ...:                                                                                                                               
In [8]: method1(x, c)                                                                                                                 
Out[8]: 
array([-50.00027239, -46.1005942 , -42.20129616, -38.30282718,
       -34.4061657 , -30.51344214, -26.62928494, -22.76370109,
       -18.93810086, -15.19726975, -11.63049992,  -8.39895618,
        -5.74097906,  -3.89357109,  -2.93248309,  -2.70671618,
        -2.96252599,  -3.48601936,  -4.14466658,  -4.86820655,
        -5.62214626,  -6.39016213,  -7.16466052,  -7.94213686,
        -8.72097964,  -9.50044909])

In [10]: def method2(x, c): 
    ...:     mult = -(c-1) * (x < 0) - 1 
    ...:     t = np.abs(x) 
    ...:     return mult * t - (c+1) * sc.log1p(np.exp(-t)) 
    ...:                                                                                                                              

In [11]: method2(x, c)                                                                                                                
Out[11]: 
array([-50.00027239, -46.1005942 , -42.20129616, -38.30282718,
       -34.4061657 , -30.51344214, -26.62928494, -22.76370109,
       -18.93810086, -15.19726975, -11.63049992,  -8.39895618,
        -5.74097906,  -3.89357109,  -2.93248309,  -2.70671618,
        -2.96252599,  -3.48601936,  -4.14466658,  -4.86820655,
        -5.62214626,  -6.39016213,  -7.16466052,  -7.94213686,
        -8.72097964,  -9.50044909])

In [12]: %timeit method1(x, c)                                                                                                        
147 µs ± 4.59 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [13]: %timeit method2(x, c)                                                                                                        
16 µs ± 156 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
+1. It might be a bit uglier, but the speedup is worth it.